### DISCUSSION TITLE
Updating dependencies with overrides when available

### DISCUSSION THREAD

**Discussion by jdalton** [renovatebot/renovate#32456] [Suggest an Idea]
### Tell us more.

Overrides in at least the JS ecosystem by way of managers like `npm`, `pnpm`, `yarn`, and `bun` are starting to come into their own. With projects like [Socket Registry](https://github.com/SocketDev/socket-registry) _(disclaimer: I'm a core dev at Socket)_ providing drop-in override replacements for packages it would be empowering to be able to subscribe to such feed updates _(maybe using [registry manifest data](https://www.npmjs.com/package/@socketsecurity/registry))_. Auto-updating dependencies and swapping out with overrides when available seem like companion features.

**Comment by rarkins** [comment]
Do you mean doing the replacements (as an alternative to the Socket CLI command), or doing updates of those replacements once they've been made, and a newer version is published?

**Reply by jdalton** [reply]
This would be as an alternative to the CLI as updates after should be handled by however Renovate handles updates `overrides`, `resolutions`, and `pnpm.overrides` fields.

To be generally applicable I image being able to specify some form of metadata that folks can use to map a dependency to an override. 

Ideally looking for a way to tie-in overrides to dependency updates because if you're bumping a version you could also potentially replace it with an override.


:bulb: Npm `overrides` are a little more tricky because if the dependency exists as a direct dependency then it needs to be replaced and a `$dep-name` needs to be used as the package spec in the override ([docs](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides)).

**Reply by jdalton** [reply]
@rarkins This may be more along the lines of [a preset](https://github.com/travi/renovate/tree/482923a0850bac79b4aa81d1aa3c58f913a20069/lib/config/presets) with [Replacements](https://github.com/renovatebot/renovate/pulls?q=is%3Apr+sort%3Aupdated-desc+author%3Atravi). Thoughts?

**Reply by rarkins** [reply]
That would work for direct dependencies as a start, but currently there's no established way for the transitive part (the need hasn't arisen before)
